zkevm-circuits/src/table.rs

```diff
@@ -517,39 +506,47 @@ impl RwTable {
             id: meta.advice_column(),
             address: meta.advice_column(),
             field_tag: meta.advice_column(),
-            storage_key_word: word::Word::new([meta.advice_column(), meta.advice_column()]),
-            value_word: word::Word::new([meta.advice_column(), meta.advice_column()]),
-            value_prev_word: word::Word::new([meta.advice_column(), meta.advice_column()]),
-            storage_key: meta.advice_column(),
-            value: meta.advice_column(),
-            value_prev: meta.advice_column(),
-            // It seems that aux1 for the moment is not using randomness
-            // TODO check in a future review
-            aux1: meta.advice_column_in(SecondPhase),
-            aux2: meta.advice_column_in(SecondPhase),
+            storage_key: word::Word::new([meta.advice_column(), meta.advice_column()]),
+            value: word::Word::new([meta.advice_column(), meta.advice_column()]),
+            value_prev: word::Word::new([meta.advice_column(), meta.advice_column()]),
+            init_val: word::Word::new([meta.advice_column(), meta.advice_column()]),
         }
     }
     fn assign<F: Field>(
         &self,
         region: &mut Region<'_, F>,
         offset: usize,
-        row: &RwRow<Value<F>>,
+        row: &RwRow<F>,
```

Comment:
If an assign function is not using `Value<F>` then it is a tech debt. Bugs like #1456 would happen if we use F in an assignment function.

Reply:
With more precisely,  consider below case
case 1: caller have `Value` in hands and calling a lib(chip) assign fn taking `F`
case 2: caller have `Value` in hands and calling a lib(chip) assign fn taking `Value`
case 3: caller have `F` in hands and calling a lib(chip) assign fn taking `F`
case 4: caller have `F` in hands and calling a lib(chip) assign fn taking `Value`

only case 1 will cause bug a bug for `assign` not called during key gen phase, since Value::unknown.map(fn) wont call. 
Case 3, 4 style will not have the bug.

Case 2 style can be our rule-of-thumb, but do you have any idea to introduce a way by `trait` + type constraints, or a lint to assure case 2 style across the whole codebase? It will be good to have comparing with manual checking